reference-api/main_test.go

```diff
@@ -22,45 +22,58 @@ func mockReleasesHandler(w http.ResponseWriter, r *http.Request) {
 	fmt.Fprint(w, `{"handler": "releases"}`)
 }
 
+// Mock release handler that simulates a 404 (to trigger commit fallback)
+func mockReleasesHandler404(w http.ResponseWriter, r *http.Request) {
+	w.WriteHeader(http.StatusNotFound)
+	fmt.Fprint(w, `Not Found`)
+}
+
 func TestUnifiedHandlerWithCache(t *testing.T) {
 	// Create an LRU cache with 5 items max (to test eviction easily)
-	cache, err := lru.NewWithEvict[string, []byte](5, onEvict)
+	cache, err := lru.NewWithEvict[string, CachedResponse](5, onEvict)
 	assert.NoError(t, err, "Failed to initialize cache")
 
-	// Mock dependencies with cache
-	deps := &HandlerDeps{
-		CommitsHandler:  mockCommitsHandler,
-		ReleasesHandler: mockReleasesHandler,
-		cache:           cache,
-	}
-
+	// Test cases
 	tests := []struct {
-		name           string
-		repo           string
-		gitRef         string
-		expectedStatus int
-		expectedBody   string
+		name             string
+		repo             string
+		gitRef           string
+		expectedStatus   int
+		expectedBody     string
+		use404Releases   bool // Whether to simulate a 404 response from ReleasesHandler
```

Comment:
Instead of making this a `bool` flag to either return a 200 or 404, maybe we can extend `mockReleasesHandler` to accept an argument of what status / body it should return so that we have more flexibility to control the mocked status and body for each test case. I could imagine needing a separate flag for each http status if we followed this pattern.